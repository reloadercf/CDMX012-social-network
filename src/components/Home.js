import { onNavigate } from '../main.js';

export const Home = () => {
  const HomeSection = document.createElement('section');
  const buttonRegister = document.createElement('button');
  const buttonLogin = document.createElement('button');
  HomeSection.classList.add('container');
  buttonLogin.classList.add('btnCTA');
  buttonRegister.classList.add('btnSecondary');
  buttonRegister.textContent = 'Regístrate';
  buttonLogin.textContent = 'Inicia sesión';

  buttonRegister.addEventListener('click', () => onNavigate('/register'));
  buttonLogin.addEventListener('click', () => onNavigate('/login'));

  HomeSection.append(buttonLogin, buttonRegister);
  return HomeSection;
};
