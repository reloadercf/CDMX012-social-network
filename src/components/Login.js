import { onNavigate } from '../main.js';
import { LoginEmail } from '../lib/auth.js';

export const Login = () => {
  const HomeSection = document.createElement('section');
  const inputEmail = document.createElement('input');
  const inputPassword = document.createElement('input');
  const buttonLogin = document.createElement('button');
  const buttonHome = document.createElement('button');
  const titleLogin = document.createElement('h3');
  HomeSection.classList.add('container');
  buttonLogin.classList.add('btnCTA');
  buttonHome.classList.add('btnSecondary');
  inputEmail.type = 'email';
  inputPassword.type = 'password';
  buttonLogin.innerText = 'Entrar';
  inputEmail.placeholder = 'Escribe tu correo';
  inputPassword.placeholder = 'Escribe tu contraseÃ±a';
  titleLogin.textContent = 'Bienvenidx al Login';
  buttonHome.textContent = 'Regresar al Home';

  buttonLogin.addEventListener('click', () => {
    LoginEmail(inputEmail.value, inputPassword.value).then((userCredential) => {
      // Signed in
      const user = userCredential.user;
      onNavigate('/wall');
      // ...
    }).catch((error) => {
      console.log(error);
    });
  });
  HomeSection.appendChild(buttonHome);
  HomeSection.append(titleLogin, inputEmail, inputPassword, buttonLogin,buttonHome);
  return HomeSection;
};
