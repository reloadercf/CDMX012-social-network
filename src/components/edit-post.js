import { updatePost } from '../lib/update-doc.js';

function saveChanges(postData, node, newData) {
  const idDoc = postData.idDocument;

  updatePost(idDoc, newData).then(() => {
    // console.log('Post edited');
    node.innerHTML = `<div class="post-content">
    <div><img src= "./assets/open-book.png" class= "book-icon"><p><strong>  ${newData.reading}</strong></p></div> <br>
    <p>${newData.text}</p>
    </div>`;
  }).catch((error) => {
    console.log(error, 'Post cannot be update');
  });
}

export function toEditable(postData, node) {
  const reading = postData.reading;
  const text = postData.text;

  node.removeChild(node.firstChild);

  const editionForm = document.createElement('form');
  editionForm.setAttribute('class', 'edit-form');
  editionForm.innerHTML = `<div class="post-content">
  <div><img src= "./assets/open-book.png" class= "book-icon"> <input type="text" class= "reading-txt" name="reading" value="${reading}"></div>
  <br>
  <textarea name="txt" class="reading-description" rows="5">${text}</textarea>
  <br>
  </div>`;

  const btnSave = document.createElement('input');
  btnSave.setAttribute('type', 'button');
  btnSave.setAttribute('id', 'btn-save');
  btnSave.setAttribute('value', 'Save');

  node.append(editionForm, btnSave);

  btnSave.addEventListener('click', (e) => {
    const readingTxt = document.querySelector('.reading-txt');
    const readingDescription = document.querySelector('.reading-description');
    const newData = {
      reading: readingTxt.value,
      text: readingDescription.value,
    };
    e.preventDefault();
    saveChanges(postData, node, newData);
  });
}
