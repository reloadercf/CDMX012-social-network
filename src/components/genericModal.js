export const genericModal = (funcion, parametros, textModal) => {
  // buscar el modal, si no existe, lo creo
  // ya con modal, le agrego la funcion y sus parametros

  // find modal or create modal
  let modal = document.querySelector('modal');
  if (modal === null) {
    modal = document.createElement('div');
    modal.setAttribute('class', 'modal show-modal');
    modal.setAttribute('id', 'modal');
    const modalContent = document.createElement('div');
    modalContent.setAttribute('class', 'modal-content');
    const closebtn = document.createElement('span');
    closebtn.setAttribute('class', 'close-button');
    closebtn.setAttribute('id', 'close-button');
    closebtn.innerHTML = 'x';
    const txtModal = document.createElement('h2');
    txtModal.innerHTML = textModal;
    const acceptbtn = document.createElement('button');
    acceptbtn.setAttribute('class', 'accept-btn');
    acceptbtn.setAttribute('id', 'accept-btn');
    acceptbtn.innerHTML = 'Yes';

    closebtn.addEventListener('click', () => {
      modal.classList.toggle('show-modal');
    });

    acceptbtn.addEventListener('click', () => {
      modal.classList.toggle('show-modal');
      funcion(...parametros);
    });

    modalContent.append(closebtn, txtModal, acceptbtn);
    modal.append(modalContent);
    document.body.appendChild(modal);
  }
};
