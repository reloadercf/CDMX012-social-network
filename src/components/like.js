/* eslint-disable max-len */
import { dislike, giveLike } from '../lib/likes.js';

export const like = (click, likeCount, userId, postId, likeList) => { // (leDiLike?, #likesDePost, uidUsuarioActual, idPost, arrayDeLikes)
  const likeElement = document.createElement('div');
  likeElement.setAttribute('class', 'like-btn');
  likeElement.setAttribute('id', 'like-btn');
  if (click === true) {
    likeElement.innerHTML = '<img src= "./assets/heart.png" alt="like fill button">';
  } else {
    likeElement.innerHTML = '<img src= "./assets/like.png" alt="like empty button">';
  }
  const count = document.createElement('span');
  count.setAttribute('class', 'likeCount');
  count.textContent = likeCount;

  likeElement.append(count);
  let clicked = click;

  likeElement.addEventListener('click', (e) => { // Asignar un eventhandler al btn like
    e.preventDefault();
    if (!clicked) {
      clicked = true; // cambia status del click
      giveLike(userId, postId, likeList); // actualizar lista de likes
      likeElement.innerHTML = '<img src= "./assets/heart.png" alt="like fill button">';
      count.textContent = parseInt(count.textContent, 10) + 1;
      likeElement.append(count);
    } else {
      dislike(userId, postId, likeList);
      clicked = false;
      likeElement.innerHTML = '<img src= "./assets/like.png" alt="like empty button">';
      count.textContent = parseInt(count.textContent, 10) - 1; // actualizar lista de likes
      likeElement.append(count);
    }
  });
  return likeElement;
};
