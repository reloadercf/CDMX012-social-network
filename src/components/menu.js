/* eslint-disable import/no-cycle */
import { onNavigate } from '../app.js';
import { signOutBR } from '../lib/auth.js';
import { getProfileInfo } from '../lib/profileUser.js';

export const menu = () => {
  const background = document.createElement('div');
  background.setAttribute('id', 'general-container');

  const menuContainer = document.createElement('div');
  menuContainer.setAttribute('id', 'menu-container');
  const profileInfo = document.createElement('div');
  profileInfo.setAttribute('id', 'profile-container');

  const close = document.createElement('img');
  close.setAttribute('src', './assets/close.png');
  close.setAttribute('id', 'close-img');

  close.addEventListener('click', () => {
    const body = document.body;
    body.style.overflow = '';
    const fathernode = document.querySelector('.readingPage');
    fathernode.removeChild(background);
  });

  const profileData = document.createElement('div');
  profileData.setAttribute('id', 'profile-data');

  getProfileInfo().then((user) => {
    profileData.innerHTML = `
        <p id="user-name">${user.name}</p>
        <p id="user-username">@${user.username}</p>
        <p id="user-bio">${user.bio}</p>
    `;
  });

  profileInfo.append(close, profileData);

  const configuration = document.createElement('div');
  configuration.setAttribute('id', 'config-container');
  const SignOut = document.createElement('div');
  const SignOutImg = document.createElement('img');
  SignOutImg.setAttribute('src', './assets/logout.png');
  const SignOutTxt = document.createElement('p');
  SignOutTxt.setAttribute('id', 'out');
  SignOutTxt.textContent = 'Sign Out';
  SignOut.append(SignOutImg, SignOutTxt);
  configuration.append(SignOut);

  SignOutTxt.addEventListener('click', () => {
    signOutBR().then(() => {
      const body = document.body;
      body.style.overflow = '';
      onNavigate('/');
    }).catch((error) => {
      console.log(error);
    });
  });

  menuContainer.append(profileInfo, configuration);
  background.append(menuContainer);

  return background;
};
