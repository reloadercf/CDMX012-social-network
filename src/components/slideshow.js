// slideshow

export const slideshow = () => {
  let slideIndex = 1;
  const slideshowContainer = document.createElement('div');
  slideshowContainer.setAttribute('class', 'slideshow-container');

  const slideOne = document.createElement('div');
  slideOne.setAttribute('class', 'slide fade');
  slideOne.innerHTML = '<img src=./assets/book1.png alt = "BookOne">';

  const slideTwo = document.createElement('div');
  slideTwo.setAttribute('class', 'slide fade');
  slideTwo.innerHTML = '<img src=./assets/book2.png alt = "BookTwo">';

  const slideThree = document.createElement('div');
  slideThree.setAttribute('class', 'slide fade');
  slideThree.innerHTML = '<img src=./assets/book3.png alt = "BookThree">';

  const slideFour = document.createElement('div');
  slideFour.setAttribute('class', 'slide fade');
  slideFour.innerHTML = '<img src=./assets/book4.png alt = "BookFour">';

  const btnPrev = document.createElement('a');
  btnPrev.setAttribute('class', 'btnPrev');
  btnPrev.setAttribute('id', 'btnPrev');
  btnPrev.innerHTML = '&#10094;';

  const btnNext = document.createElement('a');
  btnNext.setAttribute('class', 'btnNext');
  btnNext.setAttribute('id', 'btnNext');
  btnNext.innerHTML = '&#10095;';

  function showSlides(n) {
    let i;
    // const slides = document.getElementsByClassName('slide');
    const slides = [slideOne, slideTwo, slideThree, slideFour];
    if (n > slides.length) { slideIndex = 1; }
    if (n < 1) { slideIndex = slides.length; }
    for (i = 0; i < slides.length; i += 1) {
      slides[i].style.display = 'none';
    }
    slides[slideIndex - 1].style.display = 'block';
  }

  // Next/previous controls
  function plusSlides(n) {
    showSlides(slideIndex += n);
  }

  btnPrev.addEventListener('click', () => {
    plusSlides(-1);
  });

  btnNext.addEventListener('click', () => {
    plusSlides(1);
  });

  slideshowContainer.append(slideOne, slideTwo, slideThree, slideFour, btnPrev, btnNext);
  showSlides(slideIndex);
  return slideshowContainer;
};
