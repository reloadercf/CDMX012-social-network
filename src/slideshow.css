/* Slideshow container */
.slideshow-container {
    max-width: 1200px;
    position: relative;
    margin: auto;
    background-color:rgba(253, 230, 233, 0.7);
  }
  
  .slideshow-container img  {
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* Hide the images by default */
  .slide {
    display: none;
  }
  
  /* Next & previous buttons */
  .btnPrev, .btnNext {
    cursor: pointer;
    position: absolute;
    top: 50%;
    width: auto;
    margin-top: -22px;
    padding: 11px;
    background-color: #F5F5F5;
    color: #951C45;
    font-weight: bold;
    font-size: 18px;
    transition: 0.6s ease;
    box-shadow: 0 8px 8px 0 rgba(128, 128, 128, 0.295);
    border: 3px solid #951C45;
    border-radius: 0 3px 3px 0;
    user-select: none;
  }
  
  /* Position the "next button" to the right */
  .btnNext {
    right: 0;
    border-radius: 3px 0 0 3px;
  }
  
  /* On hover, add a black background color */
  .btnPrev:hover, .btnNext:hover {
    color:#F5F5F5;
    background-color: rgba(0,0,0,0.8);
  }

  
  /* Fading animation */
  .fade {
    -webkit-animation-name: fade;
    -webkit-animation-duration: 1.5s;
    animation-name: fade;
    animation-duration: 1.5s;
  }
  
  @-webkit-keyframes fade {
    from {opacity: .4}
    to {opacity: 1}
  }
  
  @keyframes fade {
    from {opacity: .4}
    to {opacity: 1}
  }