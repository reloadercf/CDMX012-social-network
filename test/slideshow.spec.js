/**
 * @jest-environment jsdom
 */

import { slideshow } from '../src/components/slideshow';

jest.mock('../src/firebase-imports.js');

describe('Tests of slideshow container', () => {
  test('render slideshow', () => {
    const renderSlideshow = slideshow();
    document.body.innerHTML = '<div id="root"></div>';
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(renderSlideshow);
    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
});

describe('Tests of slideshow container on click to plus 1', () => {
  test('render slideshow', () => {
    const renderSlideshow = slideshow();
    const btn = renderSlideshow.querySelector('#btnPrev');
    btn.dispatchEvent(new Event('click'));
    expect(btn.textContent).toBeDefined();
  });
});

describe('Tests of slideshow container on click to plus -1', () => {
  test('render slideshow', () => {
    const renderSlideshow = slideshow();
    const btn = renderSlideshow.querySelector('#btnNext');
    btn.dispatchEvent(new Event('click'));
    expect(btn.textContent).toBeDefined();
  });
});
