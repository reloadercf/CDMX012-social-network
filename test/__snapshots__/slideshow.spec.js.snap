// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of slideshow container render slideshow 1`] = `"<div class=\\"slideshow-container\\"><div class=\\"slide fade\\" style=\\"display: block;\\"><img src=\\"./assets/book1.png\\" alt=\\"BookOne\\"></div><div class=\\"slide fade\\" style=\\"display: none;\\"><img src=\\"./assets/book2.png\\" alt=\\"BookTwo\\"></div><div class=\\"slide fade\\" style=\\"display: none;\\"><img src=\\"./assets/book3.png\\" alt=\\"BookThree\\"></div><div class=\\"slide fade\\" style=\\"display: none;\\"><img src=\\"./assets/book4.png\\" alt=\\"BookFour\\"></div><a class=\\"btnPrev\\" id=\\"btnPrev\\">❮</a><a class=\\"btnNext\\" id=\\"btnNext\\">❯</a></div>"`;
